persistence/trees/module.go

```diff
@@ -0,0 +1,73 @@
+package trees
+
+import (
+	"fmt"
+
+	"github.com/pokt-network/pocket/persistence/kvstore"
+	"github.com/pokt-network/pocket/shared/modules"
+	"github.com/pokt-network/smt"
+)
+
+func (*treeStore) Create(bus modules.Bus, options ...modules.TreeStoreOption) (modules.TreeStoreModule, error) {
+	m := &treeStore{}
+
+	for _, option := range options {
+		option(m)
+	}
+
+	m.SetBus(bus)
```

Comment:
Should this be `registerModule` or just `setBus`?

Reply:
@Olshansk I'll find out more about this tomorrow with @bryanchriswhite because I need more context from him to properly answer that.

Reply:
I'll rephrase, it should be `RegisterModule` IMO.

```
func (m *bus) RegisterModule(module modules.Module) {
	module.SetBus(m)
	m.modulesRegistry.RegisterModule(module)
}
```

`SetBus`: Get access to DI
`RegisterModule`: Get access to DI and make this module accessible via DI (e.g. IBC module getting access to trees module)

Reply:
> Get access to DI and make this module accessible via DI (e.g. IBC module getting access to trees module)

Yeah I added the GetTreeStore method back to persistence module but ideally now its a sub module we want to access it straight from the bus